tests/functional/vault/test_shares.py

```diff
@@ -22,6 +27,59 @@ def test_deposit_with_wrong_amount(vault, token, gov):
         vault.deposit(balance, {"from": gov})
 
 
+def test_deposit_with_guest_list_not_invited(
+    vault, guestList, token, gov, rando, history
+):
+    # Make sure we're attempting to deposit something
+    token.transfer(rando, token.balanceOf(gov) // 2, {"from": gov})
+    balance = token.balanceOf(rando)
+    token.approve(vault, balance, {"from": rando})
+
+    # Note - don't need to call guestList.setGuests, since nobody's invited by
+    # default.
+    # gov is our bouncer
+    vault.setGuestList(guestList, {"from": gov})
+
+    # Ensure rando's not permitted to deposit
+    with brownie.reverts():
+        vault.deposit(balance, {"from": rando})
+
+    # Ensure authorized was called and that the deposit didn't revert for a
+    # different reason.
+    assert history[-1].subcalls[-1]["function"] == "authorized(address,uint256)"
+    assert history[-1].subcalls[-1]["return_value"][0] == False
+
+
+def test_deposit_with_guest_list_invited(vault, guestList, token, gov, rando, history):
+    # Make sure we're depositing something
+    token.transfer(rando, token.balanceOf(gov) // 2, {"from": gov})
+    balance = token.balanceOf(rando)
+    token.approve(vault, balance, {"from": rando})
+
+    # Allow rando into the party
+    guests = [rando]
+    invited = [True]
+    guestList.setGuests(guests, invited, {"from": gov})
+
+    # gov is our bouncer
+    vault.setGuestList(guestList, {"from": gov})
+
+    # Deposit balance
+    vault.deposit(balance, {"from": rando})
+
+    # Ensure authorized was called
+    authorizedCall = next(
+        sc
+        for sc in history[-1].subcalls
+        if sc["function"] == "authorized(address,uint256)"
+    )
+    assert authorizedCall["return_value"][0] == True
```

Comment:
This isn't really necessary, given the result of the deposit below

Reply:
The trouble is that if the guestList.authorized check in the Vault (lines 671 & 672) are removed, this call still succeeds, presenting a false positive.

Reply:
Nevermind, if I combine the test cases as you've suggested, that problem goes away since I check authorized was the cause of the revert.